ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -285,24 +371,31 @@ def set_action_for_agent(
         step.
         :param agent_group: The name of the group the agent is part of
         :param agent_id: The id of the agent the action is set for
-        :param action: A two dimensional np.ndarray corresponding to the action
+        :param action: A one dimensional np.ndarray corresponding to the action
         (either int or float)
         """
         pass
 
     @abstractmethod
-    def get_step_result(self, agent_group: AgentGroup) -> BatchedStepResult:
+    def get_result(
+        self, agent_group: AgentGroup
```

Comment:
I think this might be better as two separate calls. Or a NamedTuple with `decision_requests` and `terminal_states` as fields.

Reply:
We considered that option, but we need to make sure that the user is not going to forget to take care of the terminated agents. If they do, there could be memory leaks. This is what gym has been doing with their `step` call that returns a tuple `[observation, reward, done]`

Reply:
A NamedTuple with decision_requests and terminal_states as fields is an option, but not sure we want a named tuple with only 2 fields... Anyone else has an opinion on this?